src/common/hook/usePublishQueue/usePublishQueue.tsx

```diff
@@ -63,14 +63,25 @@ export const usePublishQueue = (
       const deferredJobs = publishingJobs.map(async (job, index) => {
         try {
           const signer = config.wallet;
-          await publishJob(job, signer);
+          if (job.contractAddress === "DNS-DID") {
+            // publish DID verifiable documents first
+            const publishedDnsDidJobs = await publishDnsDidJob(job, signer);
+            // update wrappedDocument with the signed documents
+            job.documents.forEach((document, index) => {
+              document.wrappedDocument = publishedDnsDidJobs[index];
+            });
```

Comment:
ermm, i digged into it liao this one need alot of refactoring, and if now wanna do it will be very messy, like a lot of place will have a condition to check if is DNS-DID then take from the new key, if not take from wrappedDocument. so as discussed just now?

Reply:
um ok then lidat maybe leave a comment TODO for future clarity i guess

Reply:
ok